UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/Admin/RadioNavigationalWarningEditAdminTest .cs

```diff
@@ -0,0 +1,150 @@
+using UKHO.MaritimeSafetyInformation.Common.Models.RadioNavigationalWarning.DTO;
+using UKHO.MaritimeSafetyInformation.Web.Controllers;
+using UKHO.MaritimeSafetyInformation.Web.Services;
+using UKHO.MaritimeSafetyInformation.Web.Services.Interfaces;
+
+namespace UKHO.MaritimeSafetyInformation.IntegrationTests.RadioNavigationalWarnings.Admin
+{
+    [TestFixture]
+    internal class RadioNavigationalWarningEditAdminTest : BaseRNWTest
+    {
+        public ILogger<RadioNavigationalWarningsAdminController> _fakeLogger;
+        private IRNWRepository _rnwRepository;
+        private RNWService _rnwService;
+        private TempDataDictionary _tempData;
+        private EditRadioNavigationalWarningsAdmin _fakeEditRadioNavigationalWarningsAdmin;
+        private RadioNavigationalWarningsAdminController _controller;
+
+        [OneTimeSetUp]
+        public async Task OneTimeSetUp()
+        {
+            await SeedWarningType(GetFakeWarningTypes());
+            await SeedRadioNavigationalWarnings(new List<RadioNavigationalWarning>() { GetFakeCreateRadioNavigationalWarning() });
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+            _fakeLogger = A.Fake<ILogger<RadioNavigationalWarningsAdminController>>();
+            _rnwRepository = new RNWRepository(FakeContext);
+            _rnwService = new RNWService(_rnwRepository, FakeRadioNavigationalWarningConfiguration, FakeLoggerRnwService);
+            _tempData = new(new DefaultHttpContext(), A.Fake<ITempDataProvider>());
+            _controller = new RadioNavigationalWarningsAdminController(FakeHttpContextAccessor, _fakeLogger, _rnwService);
+        }
+
+        [Test]
+        public async Task WhenCallEdit_ThenReturnViewAsync()
+        {
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            IActionResult result = await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id);
+            Assert.IsInstanceOf<IActionResult>(result);
+            Assert.IsNotNull(((ViewResult)result).ViewData["WarningType"]);
+        }
+
+        [Test]
+        public async Task WhenCallUpdateRadioNavigationWarnings_ThenRecordIsUpdated()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            IActionResult result = await _controller.Edit(9, _fakeEditRadioNavigationalWarningsAdmin);
+            Assert.IsInstanceOf<IActionResult>(result);
+            Assert.AreEqual("Record updated successfully!", _controller.TempData["message"].ToString());
+            Assert.AreEqual("Index", ((RedirectToActionResult)result).ActionName);
+            
+        }
+
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidWarningType_ThenReturnInValidDataException()
+        {
+            _controller.TempData = _tempData;
```

Comment:
Please remove unused assignment of _controller.TempData = _tempData;
Also, verify expected InvalidDataException message in Assert along with exception.

Reply:
changes done